pkg/ext-proc/handlers/server.go

```diff
@@ -105,12 +106,19 @@ func (s *Server) Process(srv extProcPb.ExternalProcessor_ProcessServer) error {
 			klog.V(logutil.DEFAULT).ErrorS(nil, "Unknown Request type", "request", v)
 			return status.Error(codes.Unknown, "unknown request type")
 		}
+
+		// This indicates error from the underlying model server.
+		if reqCtx.ResponseStatusCode != errutil.OK {
+			metrics.RecordRequestErrCounter(reqCtx.Model, reqCtx.ResolvedTargetModel, reqCtx.ResponseStatusCode)
```

Comment:
In reality, the RecordRequestErrCounter should only be recorded once per request. The request should fail upon the first error. However if there is a bug in the system,  the current implementation could record multiple times. For example, a bug in EPP could continue processing an invalid request and hitting model server error, resulting in recording an InvalidaRequest and ModelServerError.

To solve this, and improve the readability, I suggest the following:
1. Create a variable to track the error code, and only set once upon the first error. If the code tries to update the error code, we should log a warning (this indicates some bug).
2. Record the metric after the for loop, this ensures the metric is only recorded once per lifecycle of a request.

Reply:
Updated with following changes:

1. Create an error variable to handle error in the process function.
2. Report the error metric in defer function so it will only be reported once.
3. Ignore the error "Cannot receive stream request" because it will happen frequently but has no impact to the request processing. We shouldn't report an error metric for it.

I didn't make changes to "If the code tries to update the error code, we should log a warning (this indicates some bug)" because from logs this can be observed, and adding such logic might make request processing logic in the file cumbersome.